src/search_iterator.cpp

```diff
@@ -97,10 +98,28 @@ search_iterator search_iterator::operator--(int) {
     return it;
 }
 
-std::string search_iterator::get_url() const {
+std::string search_iterator::get_path() const {
     if ( ! internal ) {
         return "";
     }
+
+    std::string path = internal->get_document().get_data();
+    std::string dbDataType = internal->search->internal->database.get_metadata("data");
+    auto impl = internal->search->m_archives.at(get_fileIndex()).getImpl();
```

Comment:
The `Archive` already has a method to check if it is a new namespace scheme.
We can use it instead of getting the implementation.

(We may want to move all the internal machinery of search to use impl, but it is another change)